### PR TITLE
fix cache key in concat oneDNN kernel

### PR DESCRIPTION
**Pull request by wojtuss**
### PR types
Bug fixes

### PR changes
OPs

### Describe
This patch fixes cache key for oneDNN concat primitive so that the first dimensions of all inputs were taken into account.
It fixes the issue #31111.

### CODE REVIEW

File: paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc [hunk 1]
```diff
@@ -156,12 +167,9 @@ class ConcatMKLDNNOpKernel : public paddle::framework::OpKernel<T> {
...
 
     ConcatPrimitiveFactory<T> prim_creator;
-    // If one of the multiple inputs of concat has an input size of 0, the
-    // actual size of the multi_input will change
     std::string key = platform::CreateKey(
-        dev_ctx, paddle::framework::vectorize<int>(multi_input[0]->dims()),
-        multi_input.size(), ctx.OutputName("Out"), dt,
-        platform::ThreadIDasStr());
+        dev_ctx, GetDimsForKey(multi_input), multi_input.size(),
+        ctx.OutputName("Out"), dt, platform::ThreadIDasStr());
```
Comment (jczaja) [inline paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc]: If possible please remove platform::ThreadIDasStr() , it is added  in ExtendKeyWithThreadInfoIfNeeded().

Comment (wojtuss) [reply paddle/fluid/operators/mkldnn/concat_mkldnn_op.cc]: Done

**Review by juncaipeng — APPROVED** [review]
LGTM

**Comment by lidanqing-vv** [comment]
@wojtuss  Please cherry-pick to 2.0.2

**Comment by luotao1** [comment]
@wojtuss 

### You're Invited: Claim Your Custom PaddlePaddle Contributor Badge!
Baidu Open Source Office is looking for those of you who have quietly contributed to PaddlePaddle. We’ve prepared a special token of appreciation just for you—a custom commemorative badge engraved with your name.

image


Over the years, every step forward PaddlePaddle has taken has been made possible by your code, your passion, and your youth. Maybe you once submitted a line of code late at night, maybe you kindly answered a question in the community, or maybe it’s been a while since you last contributed…
But we’ve never forgotten you.

This time, we’re not just looking to reconnect with your code—we want to relive those years we burned brightly together.
Come join us—let’s once again—“reclaim our youth”!

If you’d like to take part, please send your **name, phone number, and a brief self-recommendation** to:
📧 [ext_paddle_oss@baidu.com](https://mailto:ext_paddle_oss@baidu.com/)
⏰ Deadline: **September 10th, 10:00 AM**
📌 Feel free to write your self-recommendation in your own style～
💡 No committee referral or repository write access is required—every past contributor is warmly welcome to join!

**Pull request merged**